README.md

```diff
@@ -695,41 +693,69 @@ This feature requires a recent compiler with complete C++20 support:
 
 If you intend to use CMake's C++ module support (and possibly Ninja), then more recent tools are required:
 
-* Visual Studio 17.4 or later (providing `cl.exe` 19.34 or later)
-* Clang 16.0.0 or later
+* Visual Studio 2022 17.4 or later (providing `cl.exe` 19.34 or later)
+* Clang 17.0.0 or later
+* GCC 14.0 or later
 * CMake 3.28 or later
 * Ninja 1.10.2 or later
 
-Either way, GCC does not completely support C++ modules, is therefore not recommended for use.
+> [!WARNING]
+> The Vulkan-Hpp C++ named module is still experimental. Some suggested ways to use it in your projects are below. The long-term goal is to submit patches to the CMake [`FindVulkan`](https://cmake.org/cmake/help/latest/module/FindVulkan.html) module so that users may transparently configure the named module, without needing to declare it as an additional library in consumer CMake code.
 
 ##### Usage with CMake
 
-CMake is recommended for use with the Vulkan C++ module, as it provides a convenient platform-agnostic way to configure your project.
-CMake version 3.28 or later is required to support C++ modules.
+CMake is recommended for use with the Vulkan-Hpp named module, as it provides a convenient platform-agnostic way to configure your project.
+CMake version 3.28 or later is required to support C++ modules. Refer to the [CMake documentation](https://cmake.org/cmake/help/latest/manual/cmake-cxxmodules.7.html) on the topic.
 
 CMake provides the [FindVulkan module](https://cmake.org/cmake/help/latest/module/FindVulkan.html), which may be used to source the Vulkan SDK and Vulkan headers on your system.
-**Note that this module does not yet provide an IMPORTED target for the Vulkan C++ module, so you must set it up manually.**
-
-To use CMake with C++ modules, you must first set up `vulkan.cppm` as the source for a library target with `FILE_SET` configured  `TYPE = CXX_MODULES`, and then link it into your project.
 
 ```cmake
 # find Vulkan SDK
 find_package( Vulkan REQUIRED )

```

Comment:
maybe we could add version requirement for the vulkan sdk that is sufficiently new to have the `*.cppm` file? Or is this too much maintenance overhead to make sure the version is sufficient for all features shown below? Probably it makes sense for a project to always require a certain version of the Vulkan SDK depending on the features they are using.

Reply:
> Probably it makes sense for a project to always require a certain version of the Vulkan SDK

Agreed.

If there's a version requirement setup, I feel this is stuff that ought to go in `vulkan_core.h` rather than here. Bunch of `#ifdef`s and `assert()`s should suffice.

Reply:
@theHamsta, you could probably use the defines `VK_HEADER_VERSION` or `VK_HEADER_VERSION_COMPLETE`. Since VulkanHpp is generated from the spec it is deployed in lock step with vulkan version in the SDK.

If you don't use the SDK and build it up your self them most likely you are already getting a specific git release or git hash, so you know if you have it already.

And if you are at top of tree well you know also.

That should cover it I think.

Reply:
But in the sample the `.cppm` from a system installation of Vulkan is used. You want then to find one that is new enough for the requirements of your project and also one that has `.cppm`

Reply:
> you could probably use the defines VK_HEADER_VERSION or VK_HEADER_VERSION_COMPLETE

@monamimani, with recent changes to Vulkan-Hpp, you could use `vk::HeaderVersion` 😉 

https://github.com/KhronosGroup/Vulkan-Hpp/blob/890e32f27b590acd64624141b7d554bf9881c4a0/vulkan/vulkan.hpp#L6830

@theHamsta, I'll add the version check in, no big deal. The sample is in a way meant to be copy-pasted, and it'd be good if the minimum example worked regardless of the version, or at least errored with an appropriate message.

Reply:
> @monamimani, with recent changes to Vulkan-Hpp, you could use vk::HeaderVersion 😉

I am on my phone, I assume you are referring to the constexpr variable. I think no ask was released with that change yet.